ui/src/app/machines/filter-nodes.js

```diff
@@ -54,15 +58,15 @@ const filterNodes = (nodes, search) => {
       // "in:" is used to filter the nodes by those that are
       // currently selected.
       terms.forEach(term => {
-        const matched = [];
-        nodes.forEach(node => {
-          if (node.$selected && term.toLowerCase() === "selected") {
-            matched.push(node);
-          } else if (!node.$selected && term.toLowerCase() === "!selected") {
-            matched.push(node);
-          }
-        });
-        nodes = matched;
+        if (term.toLowerCase() === "selected") {
+          nodes = nodes.filter(({ system_id }) =>
```

Comment:
You're redefining the function arguments, typically this is a no-no. I'm surprised the linter didn't catch it...

I probably just don't have enough context but this seems odd...So for each `term` you're redefining `nodes` with a `filter` result? So any subsequent `term` will override any previous values. Then once `terms` is exhausted it just returns that value? It looks like you could possibly return the result from a `terms.filter` on line 60.

Or you an disregard all of this if I'm way off base 😄

Reply:
Fixed (and a few bugs found along the way too).